pitop/pma/servo_motor.py

```diff
@@ -66,26 +66,19 @@ def get_angle_range(self):
 
         return (self.min_angle, self.max_angle)
 
-    def get_target_angle(self):
-        """
-        Return the current angle of the servo motor horn.
+    def get_current_angle_and_speed(self):
```

Comment:
Not sure about this. We want to use parameters as our means of getting the data (as opposed to a `get_` function). It seems to me that angle and speed should be retrieved separately, or it should return a higher level class with angle and speed properties

Reply:
I've updated everything that can be a property to be one. The get_current_angle_and_speed is useful for people who want both and since they come back from the MCU as one piece of data we should keep it here as getting the data twice using two methods (current_angle and current_speed) would be an double the data overhead (and they'd be out of sync)

Reply:
The only thing that is bothering me at the moment is we have `set_target_angle` as a normal method and we have `target_speed` as a property method - but since set_target_angle requires two arguments I'm not sure how to turn that into a property. We could use a list and have:

`servo.target_angle = [90, 50]` 

but then we force them to have to set a speed at the same time without a default value which I think is maybe worse

Suggestion: we could create a `target_angle` method as a property function that automatically calls a method `set_target_angle_and_speed` with a default speed. Then they have flexibility - I'll do that and we can see how it feels

Reply:
Good that this is using properties. Can we create a new class that has angle and speed as properties, that this returns an instance of?

Reply:
We could do something like this: https://github.com/pi-top/pi-top-Robotics-Kit-Python-Examples/blob/main/ptrobot/common/data_structures/geometry_msgs.py

But it might get confusing to people as they'll be writing something like `servo.angular.speed` and `servo.angular.angle` to get the values, which will look really similar to `servo.current_angle`

Reply:
What do you think about this?
```
# Can reflect current or target state
class ServoMotorState:
    def __init__(self, angle, speed):
        self.angle = angle
        self.speed = speed


s = ServoMotor("S1")

# Get
current_state = s.state
angle = current_state.angle
speed = current_state.speed

target_state = s.target_state
angle = target_state.angle
speed = target_state.speed


# Set target speed
s.target_speed = 5

# @target_speed.setter
# def target_speed(self, new_speed):
#     target = self.target_state
#     target.speed = new_speed
#     self.target_state = target


# Set target angle
s.target_angle = 75

# @target_angle.setter
# def target_angle(self, new_angle):
#     target = self.target_state
#     target.angle = new_angle
#     self.target_state = target


# Set target state
target_speed = 5
target_angle = 75

s.target_state = ServoMotorState(target_speed, target_angle)
```